api/CMakeLists.txt

```diff
@@ -98,6 +98,16 @@ if(WITH_ASYNC_EXPORT_PREVIEW)
   target_compile_definitions(opentelemetry_api INTERFACE ENABLE_ASYNC_EXPORT)
 endif()
 
+# A better place should be in sdk, not api
+if(WITH_OTLP_HTTP_SSL_PREVIEW)
+  target_compile_definitions(opentelemetry_api INTERFACE ENABLE_OTLP_HTTP_SSL)
```

Comment:
Is this experimental feature, as it is behind macro ?

Reply:
This is new code, so behind a macro, following the pattern for `WITH_ASYNC_EXPORT_PREVIEW` or `WITH_METRICS_EXEMPLAR_PREVIEW` when a new feature is added in a GA signal.

This helps to provide stability and mitigate risks, with optional adoption.

Once the code is merged, published in a new release, and after some time to address reported bugs if any, I hope the code can be made mainstream, by removing the feature flags ... this will be a separate PR and a separate discussion.

The good news is that the flags removal will only affect the SDK, not the API.